config/externalname.go

```diff
@@ -110,9 +110,9 @@ var NoForkExternalNameConfigs = map[string]config.ExternalName{
 	// us-west-2_abc123
 	"aws_cognito_user_pool": config.IdentifierFromProvider,
 	// auth.example.org
-	"aws_cognito_user_pool_domain": config.IdentifierFromProvider,
+	"aws_cognito_user_pool_domain": config.TemplatedStringAsIdentifier("", "{{ .parameters.domain }}"),
```

Comment:
I first tried using `config.ParameterAsIdentifier("domain")`, but that removed `domain` from the `spec.forProvider` entirely, moving it to `metadata.name`. Especially given that the domain can be a FQDN, that seemed a step too far.